### ISSUE TITLE
Some questions about details.

### ISSUE THREAD

**Issue by haoxia1** [wangchi95/CF-Font#8]
Thank you for your work. I have some questions and would like to request your help. To start, it seems that there may be some issues in the paper and code, but I am not sure if they are correct.

1. The caption of Figure 3 in the paper may have an issue, the source font should be **Kai** font instead of **Song** font.
2. https://github.com/wangchi95/CF-Font/blob/f2e5738bae61dc13c61cdb42488fcc37742e58aa/scripts/03d_cal_cf_weights.sh#L32

    it may should be `cal_cf_weight.py`

3. https://github.com/wangchi95/CF-Font/blob/f2e5738bae61dc13c61cdb42488fcc37742e58aa/scripts/04a_run_inf_cf.sh#L10

    it may should be `inf_with_style_ft_cf.py`

Below are my questions.

1. What are the selection criteria for fonts and what filtering experience is there?
2. Is the effect of training the base network completely equivalent to DGFont, that is to say, is `scripts/option_run_inf_dgfont.sh` equivalent to the inference effect of DGFont?
3. When running `sh scripts/04a_run_inf_cf.sh`, 60 * 16 characters are needed for inferring unseen fonts. Are these 16 characters used for few-shot style learning? So, does that mean there is few-shot style learning during inference? How is this done?
4. When running `sh scripts/04a_run_inf_cf.sh`, why do we need to provide 240*16 characters for inferring seen fonts? Haven't we already learned 240*800 characters in the training phase? What is the difference between this '16' and the previous one?
5. If I want to process 256*256 images, can the network be trained directly? Does the architecture need to be modified?
6. Can I use more characters (more than 16, such as 120) as style reference for inference?

I am looking forward to your reply very much. Sincerely thank you!

**Comment by wangchi95** [comment]
@haoxia1 Thanks for your correction, I will fix these typos immediately!

For other questions, I will provide answers below:
> 1. What are the selection criteria for fonts and what filtering experience is there?

Since the number of fonts is large enough to cover diverse fonts, we did not make a targeted selection of font styles, but only considered the completeness of the character set, and then eliminated some fonts (like [方正徐冰新英文书体](https://www.foundertype.com/index.php/FontInfo/index/id/333) and [拼音体](https://www.foundertype.com/index.php/FindFont/searchFont?keyword=%E6%8B%BC%E9%9F%B3)) that did not have cross-font character correspondence at all.

> 2. Is the effect of training the base network completely equivalent to DGFont, that is to say, is scripts/option_run_inf_dgfont.sh equivalent to the inference effect of DGFont?

Yes, we just made a slight change in the dataloader to speed up the inference, which has no effect on the overall inference effect. But it should be noted that our few-shot setting does not exactly correspond to DG-Font: we have tested all styles in the test set; DGFont randomly samples some styles for each character ([here](https://github.com/ecnuycxie/DG-Font/blob/b9368fd2527a42714742786a47d3bff08b8a96c8/validation/validation.py#LL82C25-L82C34)).

> 3. When running sh scripts/04a_run_inf_cf.sh, 60 * 16 characters are needed for inferring unseen fonts. Are these 16 characters used for few-shot style learning? So, does that mean there is few-shot style learning during inference? How is this done?

Yes, please refer to the _ISR (Iterative Style-vector Refinement)_ part of our paper. To put it simply, we first extract the corresponding style vector for each font, and then polish it alone with these few-shot characters in inference (remove style encoder and fix all other parts in the network, see `Fig 2.(c)` of the paper).

> 4. When running sh scripts/04a_run_inf_cf.sh, why do we need to provide 240x16 characters for inferring seen fonts? Haven't we already learned 240x800 characters in the training phase? What is the difference between this '16' and the previous one?

Yes, the 16 characters here are included in the 800 characters. The reason why we propose this input separately is mainly for the convenience of code writing (aligned with the unseen format).  At the same time, this also makes the script more general, such as in the case where few-shot fonts are not in the training set.
These few-shot characters here are used for ISR (same as the previous question), and polishing the style feature for a single font can make the model perform better on this font.

> 5. If I want to process 256*256 images, can the network be trained directly? Does the architecture need to be modified?

Yes, it can. We recommend that you make some improvements for high-resolution character images, you can refer to our baseline 1, 2, 3, 4. These are some of our previous explorations on high-resolution images, mainly referring to [DGFont++](https://arxiv.org/pdf/2212.14742.pdf). High resolution can greatly improve image details, especially calligraphy fonts.

> 6. Can I use more characters (more than 16, such as 120) as style reference for inference?

Of course you can, more style reference characters can bring a more accurate font-level style vector, resulting in a better generation effect.

**Comment by wangchi95** [comment]
Thank you @haoxia1 , the typos in scripts are fixed [here](https://github.com/wangchi95/CF-Font/commit/dfa537b7886236d4180a461cd0c81cd16df84e6c), and I will update the paper in the arxiv version.

**Comment by haoxia1** [comment]
Thank you very much for your reply. I have learned a lot. Regarding the fifth question, may I ask if I can directly generate 256*256 images and then use your network for training? If modifications are needed, what should be modified? Also, what is baseline 1,2,3,4?

**Comment by wangchi95** [comment]
> Thank you very much for your reply. I have learned a lot. Regarding the fifth question, may I ask if I can directly generate 256*256 images and then use your network for training? If modifications are needed, what should be modified? Also, what is baseline 1,2,3,4?

Yes, you can do it. We have tested with the resolution of 128x128, and CF-Font can also get a considerable improvement.

However, as described in DGFont++, DGFont still has space for improvement in the network structure. You can directly compare their network design diagrams, we tried to change the position of the FDSC module ([1,2]->[2,3]) according to DGFont++, which greatly improved the high-resolution results. You can also add its attention module, but we didn't test this.

Our CF-Font can also be further improved by replacing the base network with other networks that are more suitable for high resolution. Here we still recommend DGFont++, because it retains a high degree of freedom and does not require additional input, such as stroke decomposition information. You only need to slightly modify the generator of CFFont to get further improvement.

**Comment by haoxia1** [comment]
Thank you very much for your guidance. Wish you a happy life!

**Comment by haoxia1** [comment]
Sorry to bother you again, but if I want to modify the position of the FDSC module, does it correspond to this line of code?Is it correct to change baseline_idx to 1?

https://github.com/wangchi95/CF-Font/blob/dfa537b7886236d4180a461cd0c81cd16df84e6c/main.py#L99-L100

If not, could you please give me some guidance?

**Comment by wangchi95** [comment]
> Sorry to bother you again, but if I want to modify the position of the FDSC module, does it correspond to this line of code?Is it correct to change baseline_idx to 1?
> 
> https://github.com/wangchi95/CF-Font/blob/dfa537b7886236d4180a461cd0c81cd16df84e6c/main.py#L99-L100
> 
> If not, could you please give me some guidance?

Yes, you can modify this argument to use my setting. In addition, you may want to design this part yourself:

https://github.com/wangchi95/CF-Font/blob/dfa537b7886236d4180a461cd0c81cd16df84e6c/models/generator.py#L64-L64

**Comment by haoxia1** [comment]
> > Sorry to bother you again, but if I want to modify the position of the FDSC module, does it correspond to this line of code?Is it correct to change baseline_idx to 1?
> > https://github.com/wangchi95/CF-Font/blob/dfa537b7886236d4180a461cd0c81cd16df84e6c/main.py#L99-L100
> > 
> > If not, could you please give me some guidance?
> 
> Yes, you can modify this argument to use my setting. In addition, you may want to design this part yourself:
> 
> https://github.com/wangchi95/CF-Font/blob/dfa537b7886236d4180a461cd0c81cd16df84e6c/models/generator.py#L64-L64

Okay, thank you very much.
